### PR TITLE
Gourmet flatpak repository builder

### PR DESCRIPTION
**Pull request by OzarkShepherd**
This workflow builds a flatpak repository from the github.  The commands to install the flatpak are commented out for reference.

### CODE REVIEW

**Comment by cydanil** [comment]
I think that this workflow could be merged within the existing `CreateFlatpak`, and following that, creating a [single-file bundle](https://docs.flatpak.org/en/latest/single-file-bundles.html)

**Comment by OzarkShepherd** [comment]
The CreateFlatpack workflow is meant for building a test run of changes to Gourmet without actually installing Gourmet to a system.  The repository builder is for hosting the finished app so it can be downloaded and installed.  Merging would essentially change the function of CreateFlatpak and remove the test run.  I could do that, but is CreateFlatpak not useful by itself?

**Comment by OzarkShepherd** [comment]
The single-file bundle is now in the workflow.

File: .github/workflows/FlatpakInstallFile.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      run: |
+        cp /app/share/gourmet/ui/* /app/share/gourmet
+
+    - name: Build Repo
+      run: |
+        flatpak-builder --repo=repo --force-clean build-dir io.github.thinkle.Gourmet.yml
+
+    - name: Build Bundle
+      run: |
+        flatpak build-bundle repo Gourmet io.github.thinkle.Gourmet
```
Comment (cydanil) [inline .github/workflows/FlatpakInstallFile.yaml]: Let's make the output file `gourmet.flatpak`, so that we know what it is.

**Review by cydanil — COMMENTED** [review]
Okay, now that the bundle is created, it's possible to make it an artefact with something like 
```yaml
    - name: Upload Flatpak
      uses: actions/upload-artifact@v2
      with:
        name: gourmet.flatpak
        path: ./gourmet.flatpak
```

at the end of the workflow :)

**Comment by cydanil** [comment]
I see that the copy of files is not working:
```
  cp /app/share/gourmet/ui/* /app/share/gourmet
  shell: /usr/bin/bash -e {0}
cp: cannot stat '/app/share/gourmet/ui/*': No such file or directory
##[error]Process completed with exit code 1.
```

Maybe it can be skipped altogether, given that it's a workaround for a known bug?

**Comment by OzarkShepherd** [comment]
The workaround is taken out now.

**Comment by cydanil** [comment]
LGTM! I was able to install the flatpak from the artifact, and it installed all right on Linux Mint 20:

image

(it crashes because of #107, but's that's unrelated)

That's a massive achievement, thanks a lot for your efforts!
Absolutely cracking :)

**Review by cydanil — APPROVED** [review]
Let's leave the PR simmer a bit, would other people have some input.

In my opinion, it deprecates the previous workflow, but I have no strong opinion on that.

**Comment by cydanil** [comment]
Do you know what's the deal with the icon?

**Comment by OzarkShepherd** [comment]
I have the icons on my computer but maybe because I installed for the user only.  For some reason I thought it was fixed until I tried a system wide install in a virtual machine.  The icons I believe are not installing system wide because of permissions.  I gave the Gourmet flatpak home directory access but was reluctant to give it root file access permanently just for the one time need of installing icons.  I have asked people more proficient with flatpaks than me if there is a workaround, and am waiting for a response.

**Comment by cydanil** [comment]
In `io.github.thinkle.Gourmet.yml`, why are there two sections `PrePipDependencies` and `PipDependencies`?
I think it would be better to do a `pip install -r requirements`, which should encompass all the dependencies.

**Comment by OzarkShepherd** [comment]
The icons are a formatting issue.  They need to be renamed to io.github.thinkle.gourmet.png or .svg, which I can do in the manifest.  However, when I rename the icons for the flatpak to install, Gourmet crashes because it is looking for the old icon name.
`DEBUG:   /app/lib/python3.8/site-packages/gourmet/GourmetRecipeManager.py: 44 No RTF support
Traceback (most recent call last):
  File "/app/bin/gourmet", line 33, in <module>
    GourmetRecipeManager.startGUI()
  File "/app/lib/python3.8/site-packages/gourmet/GourmetRecipeManager.py", line 670, in startGUI
    splash.set_icon_from_file(os.path.join(icondir,'gourmet.png'))
gi.repository.GLib.Error: g-file-error-quark: Failed to open file “/app/share/icons/hicolor/48x48/apps/gourmet.png”: No such file or directory (4)
`

**Comment by cydanil** [comment]
This will help develop a more streamlined installation down the line, where we aim to add it to flathub. Thank you again for your work!

**Pull request merged**